### ISSUE TITLE
Support for filtered lists

### ISSUE THREAD

**Issue by zpratt** [seatgeek/react-infinite#103]
I have an application that I'm building where I need to be able to filter the children of the infinite list.  When nothing is filtered, everything renders fine. When the user filters, I compute whether or not an item is visible by determining if its associated item matches the filter text and then render the children accordingly. This causes the children that I'm passing to the Infinite container to be recomputed/rerendered and one that should be visible are not.

I have tried using the `containerHeight` and `useWindowAsScrollContainer` props. Both of them have issues with children disappearing. Basically, it doesn't update to show the new list of rendered children.

I'm assuming I'm overlooking something simple. Any help would be much appreciated.

**Comment by garetht** [comment]
Do you have some sample code? We have filtering implemented with React Infinite in production so it should in principle work.

**Comment by zpratt** [comment]
@garetht thank you for responding so quickly. I made a simple mistake. While experimenting with my initial solution, I overlooked a simple, but very important thing that consumers need to keep in mind. 

This component depends on having a true list of child elements that should be visible. I mistakenly was mapping over my result set, which produced an array containing both visible elements and null elements, which then caused empty children to be rendered due to the nature of how this component works. I realized my mistake and switched my `map` to a `forEach` that pushes visible elements onto an array which I then pass as children. 

This fixed my issue and I apologize for wasting your time. This component is awesome. I wrote a smoke test that loads 25k items easily.
